repos/keg-components/src/components/form/checkbox/checkbox.wrapper.js

```diff
@@ -111,18 +121,44 @@ export const CheckboxWrapper = props => {
     ...elProps
   } = props
 
+  !RightComponent && console.log({ disableCheck, disableUncheck })
+
   const [ isChecked, setChecked ] = useState(toBool(checked || value))
 
+  // ensure the handler can be fired, so long as the next check state is allowed
+  const toggleIsEnabled =
+    !disabled && (isChecked ? !disableUncheck : !disableCheck)
+
   const elThemePath =
     themePath || `form.${elType}.${(close && 'close') || 'default'}`
-  const themeStyles = useThemePath(elThemePath, styles)
-  const activeStyles = useCheckedState(isChecked, themeStyles)
+
+  const defaultStyles = useThemePath(elThemePath, styles)
+  const disabledStyles = useThemePath(`form.${elType}.disabled`, defaultStyles)
+  const activeStyles = useCheckedState(
+    isChecked,
+    toggleIsEnabled ? defaultStyles : disabledStyles
+  )
+
   const typeClassName = useThemeTypeAsClass(
     elThemePath || type,
     'keg-checkbox',
     className
   )
 
+  const defaultPressHandler = setCheckedValue(
```

Comment:
This all works but it's a bit confusing. Honestly having a hard time following it. 
May be add some notes?

Reply:
I refactored this some and added more inline documentation